src/javascripts/views/homeView.js

```diff
@@ -1,7 +1,85 @@
+import Template from "../templates/template";
 class HomeView {
     constructor() {
-        this.transactionListEl = document.querySelector(".transaction-list")
+        this.categoryEl = document.querySelector(".category-list");
         this.addBtnEl = document.querySelector(".btn-add-transaction");
+        this.summaryEl = document.querySelector(".summary");
+        this.showTransactionsBtnEl = document.querySelector(".transaction-title-text");
+        this.showSummaryBtnEl = document.querySelector(".summary-text");
+    }
+
+     //----- RENDERING -----//
+     renderTransactionList(transactions) {
+        // Create an object to store transactions by category
+        const transactionByCategory = {};
+
+        // Browse through all transactions
+        for (const transaction of transactions) {
+        // Get the category of the current transaction
+         const category = transaction.category;
+         
+        // If the category does not exist in the `transactionByCategory` object, create a new one
+        if (!transactionByCategory[category]) {
+            transactionByCategory[category] = [];
+        }
+
+        // Add the current transaction to the list of transactions of the corresponding category
+        transactionByCategory[category].push(transaction);
+    }
+
+    // Browse through categories and render corresponding transactions
+    for (const category in transactionByCategory) {
+        const transactions = transactionByCategory[category];
+        console.log(transactions);
```

Comment:
@HoangLiarPhuongLe Remove debug code

Reply:
I fixed it @hopnguyen-agilityio . Please help me check. Thanks !